src/components/[guild]/hooks/useUser.ts

```diff
@@ -1,26 +1,71 @@
 import { useWeb3React } from "@web3-react/core"
 import useKeyPair from "hooks/useKeyPair"
+import { KeyedMutator } from "swr"
 import useSWRImmutable from "swr/immutable"
 import { User } from "types"
 import { useFetcherWithSign } from "utils/fetcher"
 
-const useUser = () => {
-  const { account } = useWeb3React()
+const usePublicKey = (userId?: number) => {
+  const { id } = useUserPublic()
+
+  const idToUse = userId ?? id
+
+  const { data, mutate, error, isLoading, isValidating } = useSWRImmutable<string>(
+    idToUse ? `/v2/users/${idToUse}/public-key` : null,
+    { shouldRetryOnError: false }
+  )
+  return {
+    publicKey: data,
+    mutate,
+    error,
+    isLoading,
+    isValidating,
+  }
+}
+
+const useUser = (
+  userIdOrAddress?: number | string
+): User & { isLoading: boolean; mutate: KeyedMutator<User>; error: any } => {
+  const { id } = useUserPublic()
   const { keyPair, ready, isValid } = useKeyPair()
   const fetcherWithSign = useFetcherWithSign()
 
-  const { isLoading, data, mutate } = useSWRImmutable<User>(
-    account && ready && keyPair && isValid
-      ? [`/user/${account}`, { method: "GET", body: {} }]
+  const idToUse = userIdOrAddress ?? id
+
+  const { data, mutate, isLoading, error } = useSWRImmutable<User>(
+    idToUse && ready && keyPair && isValid
+      ? [`/v2/users/${idToUse}/profile`, { method: "GET", body: {} }]
```

Comment:
Isn't it the same as just passing the endpoint? Do we need a `{}` body for the authentication to work?

Reply:
The original idea was that we use the endpoint as the key for unauthenticated requests, and this for authenticated ones. I believe this was done this way, so we can be sure that SWR will send the authenticated request since it has a different key (maybe with the current version it is different, but I think, that when optional auth was implemented, SWR didn't refetch on function change)
In this case, however, I think we can avoid this because the URLs are different, since the public one uses address, and the authenticated one uses userId. I have to test this out, but I think it should work